src/app/error.tsx

```diff
@@ -0,0 +1,40 @@
+"use client";
+import React, { useState } from "react";
+import Button from "@mui/material/Button";
+import {
+    ErrorCenterer,
+    ErrorPanel,
+    ErrorLargeText,
+    ErrorSecondaryText,
+    ErrorRetryText,
+} from "@/app/errorPageStyling";
+
+interface ErrorProps {
+    error: Error;
+    reset: () => void;
+}
+
+const Error: React.FC<ErrorProps> = ({ error, reset }) => {
+    const [ErrorMessage, SetErrorMessage] = useState("");
+
+    const retry = (): void => {
+        SetErrorMessage("Retrying...");
+        setTimeout(() => {
+            reset();
+        }, 1000);
+    };
+    return (
+        <ErrorCenterer>
+            <ErrorPanel elevation={5}>
+                <ErrorLargeText>OOPS!</ErrorLargeText>
+                <ErrorSecondaryText>{error.message}</ErrorSecondaryText>
```

Comment:
Just thinking, what if instead of general database error, there is an error that pops out because e.g. we missed an edge case then the program crashed in runtime, rather than error at these locations
In this case, I ain't sure but either the error returns a ugly formatted message, or it doesn't return
Would you want to handle these cases too?

Reply:
From the docs: global-error.js is the least granular error UI and can be considered "catch-all" error handling for the whole application. It is unlikely to be triggered often as root components are typically less dynamic, and other error.js boundaries will catch most errors. 
I think this means global-error should catch any edge cases.